api/mfa.go

```diff
@@ -101,3 +125,62 @@ func (a *API) GenerateRecoveryCodes(w http.ResponseWriter, r *http.Request) erro
 		RecoveryCodes: recoveryCodes,
 	})
 }
+
+func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
+	const factorPrefix = "factor"
+	const imageSideLength = 300
+	ctx := r.Context()
+	user := getUser(ctx)
+	instanceID := getInstanceID(ctx)
+	if !user.MFAEnabled {
+		return MFANotEnabledError
+	}
+	params := &EnrollFactorParams{}
+	jsonDecoder := json.NewDecoder(r.Body)
+	err := jsonDecoder.Decode(params)
+	if err != nil {
+		return badRequestError("Could not read EnrollFactor params: %v", err)
+	}
+	if (params.FactorType != "totp") && (params.FactorType != "webauthn") {
+		return unprocessableEntityError("FactorType needs to be either 'totp' or 'webauthn'")
+	}
```

Comment:
i guess the implementation provided below supports `totp` for now only right? i'm not too sure if the `webauthn` provider follows the same enrollment process as `totp` but if so, then we can abstract this out into an interface where you have an `MFAProvider` that implements those generic methods. For example, something like:

```golang
type MFAProvider interface {
  Generate(...) (*EnrollFactorResponse, error)
  Verify(...) (*VerifyChallengeResponse, error)
}
```

Reply:
Yeah I think this is a good idea -- it's similar but not exactly the same

Reply:
Will add this in when I add webauthn in